src/core/platform.js

```diff
@@ -62,19 +63,22 @@ const platform = {
     name: platformName,
 
     /**
-     * String identifying the current runtime environment. Either 'browser' or 'node'.
+     * String identifying the current runtime environment. Either 'browser', 'node' or 'worker'.
      *
-     * @type {'browser' | 'node'}
+     * @type {'browser' | 'node' | 'worker'}
      */
     environment: environment,
 
     /**
      * The global object. This will be the window object when running in a browser and the global
-     * object when running in nodejs.
+     * object when running in nodejs and self when running in a worker.
      *
      * @type {object}
      */
-    global: (environment === 'browser') ? window : global,
+    global: globalThis ??
```

Comment:
Won't this throw on browsers that didn't support globalThis? You might need to do `typeof globalThis !== undefined` before attempting to reference it.

Reply:
Yes you're 100% right. Good spot. Will patch this